### PR TITLE
Multiple healthcheck improvements

### PR DESCRIPTION
**Pull request by TheMMaciek**

### CODE REVIEW

File: src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensusManager.scala [hunk 1]
```diff
@@ -688,22 +690,30 @@ class HealthCheckConsensusManager[F[_]](
...
+        peerHealthCheck.markOffline(toMarkOfflinePeers)
+      )
       // should I check the status??? I think yes as it may be that it's offline but stuck so not removed from the list and we will not add because of that
       peersToAdd = toRemain -- notOfflinePeers
       _ <- if (peersToAdd.nonEmpty)
         logger.warn(s"Peers that we don't have online but we should: ${logIds(peersToAdd.keySet)}")
       else F.unit
-      _ <- peersToAdd.toList.traverse { case (id, (_, consensus)) => addMissingPeer(id, consensus) }
+      _ <- F.start(
+        peersToAdd.toList.traverse { case (id, (_, consensus)) => addMissingPeer(id, consensus) }
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensusManager.scala:702]: can `addMissingPeer` throw? if so, the whole traverse will fail. additionally, shouldn't it be
```
traverse ( F.start ... )
```
?

File: src/main/scala/org/constellation/domain/healthcheck/HealthCheckLoggingHelper.scala [hunk 2]
```diff
@@ -109,22 +117,45 @@ object HealthCheckLoggingHelper {
 
   // Logging and metrics
   def logHealthcheckConsensusDecicion[F[_]: Sync](
-    decision: HealthcheckConsensusDecision,
+    decision: HealthcheckConsensusDecision)(
+    metrics: Metrics,
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/domain/healthcheck/HealthCheckLoggingHelper.scala:121]: why currying if it's not used?

File: src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensusManager.scala (hunk 1 above)
Comment (TheMMaciek) [reply src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensusManager.scala:702]: All three methods: addMissingPeer, markOfflinePeer, notifyPeerAboutMissedConsensus have the error handling as the last step of the flow, so they shouldn't throw and hence interrupt the traversals I think.

File: src/main/scala/org/constellation/domain/healthcheck/HealthCheckLoggingHelper.scala (hunk 2 above)
Comment (TheMMaciek) [reply src/main/scala/org/constellation/domain/healthcheck/HealthCheckLoggingHelper.scala:121]: Separation of data (decision) and services that are used to do something with that data (metrics, logger). I'm not using currying, but I could and I think it looks cleaner.

**Review by tbekas — APPROVED** [review]
Looks good to me!

File: src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala [hunk 3]
```diff
@@ -116,15 +116,18 @@ class HealthCheckConsensus[F[_]](
...
+    for {
+      _ <- metrics.incrementMetricAsync[F](s"healthcheck_peerRemovedFromConsensus_$removalReason")
+      _ <- removedPeers.modify { alreadyRemoved =>
+        val updatedRemovalReasons =
+          alreadyRemoved
+            .get(id)
+            .map(_ :+ removalReason)
+            .getOrElse(NonEmptyList.one(removalReason))
+
+        (alreadyRemoved + (id -> updatedRemovalReasons), ())
```
Comment (kpudlik) [inline src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala:128]: JFYI there is `updatedWith` method which makes `get` and `+ (k -> v)` at once:

```scala
val updatedRemovalReasons = 
  alreadyRemoved.updatedWith(id) {
     _.map(_ :+ removalReason).orElse(NonEmptyList.one(removalReason).some)
  }
  
(updatedRemovalReasons, ())
```

**Pull request merged**